addons/pos_adyen/controllers/main.py

```diff
@@ -0,0 +1,33 @@
+# coding: utf-8
+import logging
+import pprint
+import json
+from odoo import fields, http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class PosAdyenController(http.Controller):
+    @http.route('/pos_adyen/notification', type='json', methods=['POST'], auth='none', csrf=False)
+    def notification(self):
+        data = json.loads(request.httprequest.data)
+        _logger.info('notification received from adyen:\n%s', pprint.pformat(data))
+
+        # ignore if it's not a response to a sales request
+        if not data.get('SaleToPOIResponse'):
+            _logger.info('ignoring notification because it\'s not a SaleToPOIResponse')
+            return
+
+        terminal_identifier = data['SaleToPOIResponse']['MessageHeader']['POIID']
+        payment_method = request.env['pos.payment.method'].sudo().search([('adyen_terminal_identifier', '=', terminal_identifier)], limit=1)
```

Comment:
I think it will cause trouble, if the same terminal is set on multiple payment_method, so we can either put a constraint on the pos payment method to avoid having two with the same terminal_identifier. Or we can set the response on all payment method with the identifier, (but we'll have to clear all 'latest_response' when it's used from one payment method)

Reply:
Yes indeed, thanks! I thought of this case but then forgot to do something about it. I added a constraint on `adyen_terminal_identifier` to prevent sharing terminals between different POSes. I think trying to share them would be too confusing anyway.